### ISSUE TITLE
Detect final methods in Enumeration for RedundantModifier

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#2068]
Code:

```
package test;

public enum TestEnum1 {
    ;

    public static final TestEnum1 get() { // line 6
        return null;
    }

    public final TestEnum1 test() { // line 10
        return null;
    }
}
```

Config:

```
<module name="Checker">
    <property name="charset" value="ASCII" />
    <property name="severity" value="error" />
    <property name="fileExtensions" value="java" />

    <module name="TreeWalker">
        <module name="RedundantModifier">
        </module>
    </module>
</module>
```

Output:

```
Starting audit...
Audit done.
```

I expected line 6's and 10's "final" to be marked as a redundant modifier but it is not. Final methods mean it cannot be overridden when the class is inherited but enumerations cannot be inherited anyways as they are not a class. Check style should treat the enumeration as if it were a final class, which it would then mark as redundant.
I am using checkstyle 6.10.1 .
##

**Comment by tgyurci** [comment]
Enum members can be subclasses of the enum type and some member can override the base enum class's method:
```java
public enum TestEnum {
	E1,

	E2 {
		@Override
		public void v() {
		}
	};

	public void v() {
	}
}
```

That's why it makes sense that enum member functions are final or not. In addition the RedundantModifierCheck marks `final` redunant in enum member declarations:

```java
public enum TestEnum {
	E1,

	E2 {
		@Override
		public final void v() {
		}
	};

	public void v() {
	}
}
```

**Comment by romani** [comment]
We need to extend input code base with example from tgyurci . It is not obvious.

**Comment by rnveach** [comment]
> RedundantModifierCheck marks final redunant in enum member declarations

I wasn't getting any violations on current version or previous versions, so I will create a new issue for this.

**Comment by romani** [comment]
Fix is merged
